File: src/backend/vendas/src/model/venda_cabecalho_model.py
from src import db
from src.model.venda_comissao_model import VendaComissaoModel
from src.model.venda_detalhe_model import VendaDetalheModel
from src.model.venda_frete_model import VendaFreteModel
from src.model.venda_condicoes_pagamento_model import VendaCondicoesPagamentoModel
from src.model.view_pessoa_vendedor_model import ViewPessoaVendedorModel
from src.model.view_pessoa_transportadora_model import ViewPessoaTransportadoraModel
from src.model.view_pessoa_cliente_model import ViewPessoaClienteModel
from src.model.venda_orcamento_cabecalho_model import VendaOrcamentoCabecalhoModel
from src.model.nota_fiscal_tipo_model import NotaFiscalTipoModel


class VendaCabecalhoModel(db.Model):
    __tablename__ = 'venda_cabecalho'

    id = db.Column(db.Integer, primary_key=True)
    local_entrega = db.Column(db.String(100))
    local_cobranca = db.Column(db.String(100))
    tipo_frete = db.Column(db.String(1))
    forma_pagamento = db.Column(db.String(1))
    data_venda = db.Column(db.DateTime)
    data_saida = db.Column(db.DateTime)
    hora_saida = db.Column(db.String(8))
    numero_fatura = db.Column(db.Integer)
    valor_frete = db.Column(db.Float)
    valor_seguro = db.Column(db.Float)
    valor_subtotal = db.Column(db.Float)
    taxa_comissao = db.Column(db.Float)
    valor_comissao = db.Column(db.Float)
    taxa_desconto = db.Column(db.Float)
    valor_desconto = db.Column(db.Float)
    valor_total = db.Column(db.Float)
    situacao = db.Column(db.String(1))
    dia_fixo_parcela = db.Column(db.String(2))
    observacao = db.Column(db.Text)
    id_venda_condicoes_pagamento = db.Column(db.Integer, db.ForeignKey('venda_condicoes_pagamento.id'))
    id_vendedor = db.Column(db.Integer, db.ForeignKey('view_pessoa_vendedor.id'))
    id_transportadora = db.Column(db.Integer, db.ForeignKey('view_pessoa_transportadora.id'))
    id_cliente = db.Column(db.Integer, db.ForeignKey('view_pessoa_cliente.id'))
    id_venda_orcamento_cabecalho = db.Column(db.Integer, db.ForeignKey('venda_orcamento_cabecalho.id'))
    id_nota_fiscal_tipo = db.Column(db.Integer, db.ForeignKey('nota_fiscal_tipo.id'))

    venda_comissao_model = db.relationship('VendaComissaoModel', uselist=False)
    venda_detalhe_model_list = db.relationship('VendaDetalheModel', lazy='dynamic')
    venda_frete_model_list = db.relationship('VendaFreteModel', lazy='dynamic')
    venda_condicoes_pagamento_model = db.relationship('VendaCondicoesPagamentoModel', foreign_keys=[id_venda_condicoes_pagamento])
    view_pessoa_vendedor_model = db.relationship('ViewPessoaVendedorModel', foreign_keys=[id_vendedor])
    view_pessoa_transportadora_model = db.relationship('ViewPessoaTransportadoraModel', foreign_keys=[id_transportadora])
    view_pessoa_cliente_model = db.relationship('ViewPessoaClienteModel', foreign_keys=[id_cliente])
    venda_orcamento_cabecalho_model = db.relationship('VendaOrcamentoCabecalhoModel', foreign_keys=[id_venda_orcamento_cabecalho])
    nota_fiscal_tipo_model = db.relationship('NotaFiscalTipoModel', foreign_keys=[id_nota_fiscal_tipo])

    def mapping(self, data):
        self.id = data.get('id')
        self.id_venda_orcamento_cabecalho = data.get('idVendaOrcamentoCabecalho')
        self.id_nota_fiscal_tipo = data.get('idNotaFiscalTipo')
        self.id_vendedor = data.get('idVendedor')
        self.id_venda_condicoes_pagamento = data.get('idVendaCondicoesPagamento')
        self.id_transportadora = data.get('idTransportadora')
        self.id_cliente = data.get('idCliente')
        self.local_entrega = data.get('localEntrega')
        self.local_cobranca = data.get('localCobranca')
        self.tipo_frete = data.get('tipoFrete')
        self.forma_pagamento = data.get('formaPagamento')
        self.data_venda = data.get('dataVenda')
        self.data_saida = data.get('dataSaida')
        self.hora_saida = data.get('horaSaida')
        self.numero_fatura = data.get('numeroFatura')
        self.valor_frete = data.get('valorFrete')
        self.valor_seguro = data.get('valorSeguro')
        self.valor_subtotal = data.get('valorSubtotal')
        self.taxa_comissao = data.get('taxaComissao')
        self.valor_comissao = data.get('valorComissao')
        self.taxa_desconto = data.get('taxaDesconto')
        self.valor_desconto = data.get('valorDesconto')
        self.valor_total = data.get('valorTotal')
        self.situacao = data.get('situacao')
        self.dia_fixo_parcela = data.get('diaFixoParcela')
        self.observacao = data.get('observacao')

    def serialize(self):
        return {
            'id': self.id,
            'idVendaOrcamentoCabecalho': self.id_venda_orcamento_cabecalho,
            'idNotaFiscalTipo': self.id_nota_fiscal_tipo,
            'idVendedor': self.id_vendedor,
            'idVendaCondicoesPagamento': self.id_venda_condicoes_pagamento,
            'idTransportadora': self.id_transportadora,
            'idCliente': self.id_cliente,
            'localEntrega': self.local_entrega,
            'localCobranca': self.local_cobranca,
            'tipoFrete': self.tipo_frete,
            'formaPagamento': self.forma_pagamento,
            'dataVenda': self.data_venda.isoformat(),
            'dataSaida': self.data_saida.isoformat(),
            'horaSaida': self.hora_saida,
            'numeroFatura': self.numero_fatura,
            'valorFrete': self.valor_frete,
            'valorSeguro': self.valor_seguro,
            'valorSubtotal': self.valor_subtotal,
            'taxaComissao': self.taxa_comissao,
            'valorComissao': self.valor_comissao,
            'taxaDesconto': self.taxa_desconto,
            'valorDesconto': self.valor_desconto,
            'valorTotal': self.valor_total,
            'situacao': self.situacao,
            'diaFixoParcela': self.dia_fixo_parcela,
            'observacao': self.observacao,
            'vendaComissaoModel': self.venda_comissao_model.serialize() if self.venda_comissao_model else None,
            'vendaDetalheModelList': [venda_detalhe_model.serialize() for venda_detalhe_model in self.venda_detalhe_model_list],
            'vendaFreteModelList': [venda_frete_model.serialize() for venda_frete_model in self.venda_frete_model_list],
            'vendaCondicoesPagamentoModel': self.venda_condicoes_pagamento_model.serialize() if self.venda_condicoes_pagamento_model else None,
            'viewPessoaVendedorModel': self.view_pessoa_vendedor_model.serialize() if self.view_pessoa_vendedor_model else None,
            'viewPessoaTransportadoraModel': self.view_pessoa_transportadora_model.serialize() if self.view_pessoa_transportadora_model else None,
            'viewPessoaClienteModel': self.view_pessoa_cliente_model.serialize() if self.view_pessoa_cliente_model else None,
            'vendaOrcamentoCabecalhoModel': self.venda_orcamento_cabecalho_model.serialize() if self.venda_orcamento_cabecalho_model else None,
            'notaFiscalTipoModel': self.nota_fiscal_tipo_model.serialize() if self.nota_fiscal_tipo_model else None,
        }

File: src/frontend/vendas/lib/app/data/model/venda_cabecalho_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:vendas/app/infra/infra_imports.dart';
import 'package:vendas/app/data/model/model_imports.dart';
import 'package:intl/intl.dart';
import 'package:vendas/app/data/domain/domain_imports.dart';

class VendaCabecalhoModel {
	int? id;
	int? idVendaOrcamentoCabecalho;
	int? idNotaFiscalTipo;
	int? idVendedor;
	int? idVendaCondicoesPagamento;
	int? idTransportadora;
	int? idCliente;
	String? localEntrega;
	String? localCobranca;
	String? tipoFrete;
	String? formaPagamento;
	DateTime? dataVenda;
	DateTime? dataSaida;
	String? horaSaida;
	int? numeroFatura;
	double? valorFrete;
	double? valorSeguro;
	double? valorSubtotal;
	double? taxaComissao;
	double? valorComissao;
	double? taxaDesconto;
	double? valorDesconto;
	double? valorTotal;
	String? situacao;
	String? diaFixoParcela;
	String? observacao;
	VendaComissaoModel? vendaComissaoModel;
	List<VendaDetalheModel>? vendaDetalheModelList;
	List<VendaFreteModel>? vendaFreteModelList;
	VendaCondicoesPagamentoModel? vendaCondicoesPagamentoModel;
	ViewPessoaVendedorModel? viewPessoaVendedorModel;
	ViewPessoaTransportadoraModel? viewPessoaTransportadoraModel;
	ViewPessoaClienteModel? viewPessoaClienteModel;
	VendaOrcamentoCabecalhoModel? vendaOrcamentoCabecalhoModel;
	NotaFiscalTipoModel? notaFiscalTipoModel;

	VendaCabecalhoModel({
		this.id,
		this.idVendaOrcamentoCabecalho,
		this.idNotaFiscalTipo,
		this.idVendedor,
		this.idVendaCondicoesPagamento,
		this.idTransportadora,
		this.idCliente,
		this.localEntrega,
		this.localCobranca,
		this.tipoFrete,
		this.formaPagamento,
		this.dataVenda,
		this.dataSaida,
		this.horaSaida,
		this.numeroFatura,
		this.valorFrete,
		this.valorSeguro,
		this.valorSubtotal,
		this.taxaComissao,
		this.valorComissao,
		this.taxaDesconto,
		this.valorDesconto,
		this.valorTotal,
		this.situacao,
		this.diaFixoParcela,
		this.observacao,
		this.vendaComissaoModel,
		this.vendaDetalheModelList,
		this.vendaFreteModelList,
		this.vendaCondicoesPagamentoModel,
		this.viewPessoaVendedorModel,
		this.viewPessoaTransportadoraModel,
		this.viewPessoaClienteModel,
		this.vendaOrcamentoCabecalhoModel,
		this.notaFiscalTipoModel,
	});

	static List<String> dbColumns = <String>[
		'id',
		'local_entrega',
		'local_cobranca',
		'tipo_frete',
		'forma_pagamento',
		'data_venda',
		'data_saida',
		'hora_saida',
		'numero_fatura',
		'valor_frete',
		'valor_seguro',
		'valor_subtotal',
		'taxa_comissao',
		'valor_comissao',
		'taxa_desconto',
		'valor_desconto',
		'valor_total',
		'situacao',
		'dia_fixo_parcela',
		'observacao',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Local Entrega',
		'Local Cobranca',
		'Tipo Frete',
		'Forma Pagamento',
		'Data Venda',
		'Data Saida',
		'Hora Saida',
		'Numero Fatura',
		'Valor Frete',
		'Valor Seguro',
		'Valor Subtotal',
		'Taxa Comissao',
		'Valor Comissao',
		'Taxa Desconto',
		'Valor Desconto',
		'Valor Total',
		'Situacao',
		'Dia Fixo Parcela',
		'Observacao',
	];

	VendaCabecalhoModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idVendaOrcamentoCabecalho = jsonData['idVendaOrcamentoCabecalho'];
		idNotaFiscalTipo = jsonData['idNotaFiscalTipo'];
		idVendedor = jsonData['idVendedor'];
		idVendaCondicoesPagamento = jsonData['idVendaCondicoesPagamento'];
		idTransportadora = jsonData['idTransportadora'];
		idCliente = jsonData['idCliente'];
		localEntrega = jsonData['localEntrega'];
		localCobranca = jsonData['localCobranca'];
		tipoFrete = VendaCabecalhoDomain.getTipoFrete(jsonData['tipoFrete']);
		formaPagamento = VendaCabecalhoDomain.getFormaPagamento(jsonData['formaPagamento']);
		dataVenda = jsonData['dataVenda'] != null ? DateTime.tryParse(jsonData['dataVenda']) : null;
		dataSaida = jsonData['dataSaida'] != null ? DateTime.tryParse(jsonData['dataSaida']) : null;
		horaSaida = jsonData['horaSaida'];
		numeroFatura = jsonData['numeroFatura'];
		valorFrete = jsonData['valorFrete']?.toDouble();
		valorSeguro = jsonData['valorSeguro']?.toDouble();
		valorSubtotal = jsonData['valorSubtotal']?.toDouble();
		taxaComissao = jsonData['taxaComissao']?.toDouble();
		valorComissao = jsonData['valorComissao']?.toDouble();
		taxaDesconto = jsonData['taxaDesconto']?.toDouble();
		valorDesconto = jsonData['valorDesconto']?.toDouble();
		valorTotal = jsonData['valorTotal']?.toDouble();
		situacao = VendaCabecalhoDomain.getSituacao(jsonData['situacao']);
		diaFixoParcela = jsonData['diaFixoParcela'];
		observacao = jsonData['observacao'];
		vendaComissaoModel = jsonData['vendaComissaoModel'] == null ? VendaComissaoModel(viewPessoaVendedorModel: ViewPessoaVendedorModel(), ) : VendaComissaoModel.fromJson(jsonData['vendaComissaoModel']);
		vendaDetalheModelList = (jsonData['vendaDetalheModelList'] as Iterable?)?.map((m) => VendaDetalheModel.fromJson(m)).toList() ?? [];
		vendaFreteModelList = (jsonData['vendaFreteModelList'] as Iterable?)?.map((m) => VendaFreteModel.fromJson(m)).toList() ?? [];
		vendaCondicoesPagamentoModel = jsonData['vendaCondicoesPagamentoModel'] == null ? VendaCondicoesPagamentoModel() : VendaCondicoesPagamentoModel.fromJson(jsonData['vendaCondicoesPagamentoModel']);
		viewPessoaVendedorModel = jsonData['viewPessoaVendedorModel'] == null ? ViewPessoaVendedorModel() : ViewPessoaVendedorModel.fromJson(jsonData['viewPessoaVendedorModel']);
		viewPessoaTransportadoraModel = jsonData['viewPessoaTransportadoraModel'] == null ? ViewPessoaTransportadoraModel() : ViewPessoaTransportadoraModel.fromJson(jsonData['viewPessoaTransportadoraModel']);
		viewPessoaClienteModel = jsonData['viewPessoaClienteModel'] == null ? ViewPessoaClienteModel() : ViewPessoaClienteModel.fromJson(jsonData['viewPessoaClienteModel']);
		vendaOrcamentoCabecalhoModel = jsonData['vendaOrcamentoCabecalhoModel'] == null ? VendaOrcamentoCabecalhoModel() : VendaOrcamentoCabecalhoModel.fromJson(jsonData['vendaOrcamentoCabecalhoModel']);
		notaFiscalTipoModel = jsonData['notaFiscalTipoModel'] == null ? NotaFiscalTipoModel() : NotaFiscalTipoModel.fromJson(jsonData['notaFiscalTipoModel']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idVendaOrcamentoCabecalho'] = idVendaOrcamentoCabecalho != 0 ? idVendaOrcamentoCabecalho : null;
		jsonData['idNotaFiscalTipo'] = idNotaFiscalTipo != 0 ? idNotaFiscalTipo : null;
		jsonData['idVendedor'] = idVendedor != 0 ? idVendedor : null;
		jsonData['idVendaCondicoesPagamento'] = idVendaCondicoesPagamento != 0 ? idVendaCondicoesPagamento : null;
		jsonData['idTransportadora'] = idTransportadora != 0 ? idTransportadora : null;
		jsonData['idCliente'] = idCliente != 0 ? idCliente : null;
		jsonData['localEntrega'] = localEntrega;
		jsonData['localCobranca'] = localCobranca;
		jsonData['tipoFrete'] = VendaCabecalhoDomain.setTipoFrete(tipoFrete);
		jsonData['formaPagamento'] = VendaCabecalhoDomain.setFormaPagamento(formaPagamento);
		jsonData['dataVenda'] = dataVenda != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataVenda!) : null;
		jsonData['dataSaida'] = dataSaida != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataSaida!) : null;
		jsonData['horaSaida'] = Util.removeMask(horaSaida);
		jsonData['numeroFatura'] = numeroFatura;
		jsonData['valorFrete'] = valorFrete;
		jsonData['valorSeguro'] = valorSeguro;
		jsonData['valorSubtotal'] = valorSubtotal;
		jsonData['taxaComissao'] = taxaComissao;
		jsonData['valorComissao'] = valorComissao;
		jsonData['taxaDesconto'] = taxaDesconto;
		jsonData['valorDesconto'] = valorDesconto;
		jsonData['valorTotal'] = valorTotal;
		jsonData['situacao'] = VendaCabecalhoDomain.setSituacao(situacao);
		jsonData['diaFixoParcela'] = diaFixoParcela;
		jsonData['observacao'] = observacao;
		jsonData['vendaComissaoModel'] = vendaComissaoModel?.toJson;
		
		var vendaDetalheModelLocalList = []; 
		for (VendaDetalheModel object in vendaDetalheModelList ?? []) { 
			vendaDetalheModelLocalList.add(object.toJson); 
		}
		jsonData['vendaDetalheModelList'] = vendaDetalheModelLocalList;
		
		var vendaFreteModelLocalList = []; 
		for (VendaFreteModel object in vendaFreteModelList ?? []) { 
			vendaFreteModelLocalList.add(object.toJson); 
		}
		jsonData['vendaFreteModelList'] = vendaFreteModelLocalList;
		jsonData['vendaCondicoesPagamentoModel'] = vendaCondicoesPagamentoModel?.toJson;
		jsonData['viewPessoaVendedorModel'] = viewPessoaVendedorModel?.toJson;
		jsonData['viewPessoaTransportadoraModel'] = viewPessoaTransportadoraModel?.toJson;
		jsonData['viewPessoaClienteModel'] = viewPessoaClienteModel?.toJson;
		jsonData['vendaOrcamentoCabecalhoModel'] = vendaOrcamentoCabecalhoModel?.toJson;
		jsonData['notaFiscalTipoModel'] = notaFiscalTipoModel?.toJson;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idVendaOrcamentoCabecalho = plutoRow.cells['idVendaOrcamentoCabecalho']?.value;
		idNotaFiscalTipo = plutoRow.cells['idNotaFiscalTipo']?.value;
		idVendedor = plutoRow.cells['idVendedor']?.value;
		idVendaCondicoesPagamento = plutoRow.cells['idVendaCondicoesPagamento']?.value;
		idTransportadora = plutoRow.cells['idTransportadora']?.value;
		idCliente = plutoRow.cells['idCliente']?.value;
		localEntrega = plutoRow.cells['localEntrega']?.value;
		localCobranca = plutoRow.cells['localCobranca']?.value;
		tipoFrete = plutoRow.cells['tipoFrete']?.value != '' ? plutoRow.cells['tipoFrete']?.value : 'CIF';
		formaPagamento = plutoRow.cells['formaPagamento']?.value != '' ? plutoRow.cells['formaPagamento']?.value : 'A Vista';
		dataVenda = Util.stringToDate(plutoRow.cells['dataVenda']?.value);
		dataSaida = Util.stringToDate(plutoRow.cells['dataSaida']?.value);
		horaSaida = plutoRow.cells['horaSaida']?.value;
		numeroFatura = plutoRow.cells['numeroFatura']?.value;
		valorFrete = plutoRow.cells['valorFrete']?.value?.toDouble();
		valorSeguro = plutoRow.cells['valorSeguro']?.value?.toDouble();
		valorSubtotal = plutoRow.cells['valorSubtotal']?.value?.toDouble();
		taxaComissao = plutoRow.cells['taxaComissao']?.value?.toDouble();
		valorComissao = plutoRow.cells['valorComissao']?.value?.toDouble();
		taxaDesconto = plutoRow.cells['taxaDesconto']?.value?.toDouble();
		valorDesconto = plutoRow.cells['valorDesconto']?.value?.toDouble();
		valorTotal = plutoRow.cells['valorTotal']?.value?.toDouble();
		situacao = plutoRow.cells['situacao']?.value != '' ? plutoRow.cells['situacao']?.value : 'Digitação';
		diaFixoParcela = plutoRow.cells['diaFixoParcela']?.value;
		observacao = plutoRow.cells['observacao']?.value;
		vendaComissaoModel = VendaComissaoModel(viewPessoaVendedorModel: ViewPessoaVendedorModel(), );
		vendaDetalheModelList = [];
		vendaFreteModelList = [];
		vendaCondicoesPagamentoModel = VendaCondicoesPagamentoModel();
		vendaCondicoesPagamentoModel?.nome = plutoRow.cells['vendaCondicoesPagamentoModel']?.value;
		viewPessoaVendedorModel = ViewPessoaVendedorModel();
		viewPessoaVendedorModel?.nome = plutoRow.cells['viewPessoaVendedorModel']?.value;
		viewPessoaTransportadoraModel = ViewPessoaTransportadoraModel();
		viewPessoaTransportadoraModel?.nome = plutoRow.cells['viewPessoaTransportadoraModel']?.value;
		viewPessoaClienteModel = ViewPessoaClienteModel();
		viewPessoaClienteModel?.nome = plutoRow.cells['viewPessoaClienteModel']?.value;
		vendaOrcamentoCabecalhoModel = VendaOrcamentoCabecalhoModel();
		vendaOrcamentoCabecalhoModel?.codigo = plutoRow.cells['vendaOrcamentoCabecalhoModel']?.value;
		notaFiscalTipoModel = NotaFiscalTipoModel();
		notaFiscalTipoModel?.nome = plutoRow.cells['notaFiscalTipoModel']?.value;
	}	

	VendaCabecalhoModel clone() {
		return VendaCabecalhoModel(
			id: id,
			idVendaOrcamentoCabecalho: idVendaOrcamentoCabecalho,
			idNotaFiscalTipo: idNotaFiscalTipo,
			idVendedor: idVendedor,
			idVendaCondicoesPagamento: idVendaCondicoesPagamento,
			idTransportadora: idTransportadora,
			idCliente: idCliente,
			localEntrega: localEntrega,
			localCobranca: localCobranca,
			tipoFrete: tipoFrete,
			formaPagamento: formaPagamento,
			dataVenda: dataVenda,
			dataSaida: dataSaida,
			horaSaida: horaSaida,
			numeroFatura: numeroFatura,
			valorFrete: valorFrete,
			valorSeguro: valorSeguro,
			valorSubtotal: valorSubtotal,
			taxaComissao: taxaComissao,
			valorComissao: valorComissao,
			taxaDesconto: taxaDesconto,
			valorDesconto: valorDesconto,
			valorTotal: valorTotal,
			situacao: situacao,
			diaFixoParcela: diaFixoParcela,
			observacao: observacao,
			vendaComissaoModel: VendaComissaoModel(
				id: vendaComissaoModel?.id,
				idVendaCabecalho: vendaComissaoModel?.idVendaCabecalho,
				idVendedor: vendaComissaoModel?.idVendedor,
				valorVenda: vendaComissaoModel?.valorVenda,
				tipoContabil: vendaComissaoModel?.tipoContabil,
				valorComissao: vendaComissaoModel?.valorComissao,
				situacao: vendaComissaoModel?.situacao,
				dataLancamento: vendaComissaoModel?.dataLancamento,
				viewPessoaVendedorModel: ViewPessoaVendedorModel(
					id: vendaComissaoModel?.viewPessoaVendedorModel?.id,
					nome: vendaComissaoModel?.viewPessoaVendedorModel?.nome,
					tipo: vendaComissaoModel?.viewPessoaVendedorModel?.tipo,
					email: vendaComissaoModel?.viewPessoaVendedorModel?.email,
					site: vendaComissaoModel?.viewPessoaVendedorModel?.site,
					cpfCnpj: vendaComissaoModel?.viewPessoaVendedorModel?.cpfCnpj,
					rgIe: vendaComissaoModel?.viewPessoaVendedorModel?.rgIe,
					matricula: vendaComissaoModel?.viewPessoaVendedorModel?.matricula,
					dataCadastro: vendaComissaoModel?.viewPessoaVendedorModel?.dataCadastro,
					dataAdmissao: vendaComissaoModel?.viewPessoaVendedorModel?.dataAdmissao,
					dataDemissao: vendaComissaoModel?.viewPessoaVendedorModel?.dataDemissao,
					ctpsNumero: vendaComissaoModel?.viewPessoaVendedorModel?.ctpsNumero,
					ctpsSerie: vendaComissaoModel?.viewPessoaVendedorModel?.ctpsSerie,
					ctpsDataExpedicao: vendaComissaoModel?.viewPessoaVendedorModel?.ctpsDataExpedicao,
					ctpsUf: vendaComissaoModel?.viewPessoaVendedorModel?.ctpsUf,
					observacao: vendaComissaoModel?.viewPessoaVendedorModel?.observacao,
					logradouro: vendaComissaoModel?.viewPessoaVendedorModel?.logradouro,
					numero: vendaComissaoModel?.viewPessoaVendedorModel?.numero,
					complemento: vendaComissaoModel?.viewPessoaVendedorModel?.complemento,
					bairro: vendaComissaoModel?.viewPessoaVendedorModel?.bairro,
					cidade: vendaComissaoModel?.viewPessoaVendedorModel?.cidade,
					cep: vendaComissaoModel?.viewPessoaVendedorModel?.cep,
					municipioIbge: vendaComissaoModel?.viewPessoaVendedorModel?.municipioIbge,
					uf: vendaComissaoModel?.viewPessoaVendedorModel?.uf,
					idPessoa: vendaComissaoModel?.viewPessoaVendedorModel?.idPessoa,
					idCargo: vendaComissaoModel?.viewPessoaVendedorModel?.idCargo,
					idSetor: vendaComissaoModel?.viewPessoaVendedorModel?.idSetor,
					comissao: vendaComissaoModel?.viewPessoaVendedorModel?.comissao,
					metaVenda: vendaComissaoModel?.viewPessoaVendedorModel?.metaVenda,
				),
			),
			vendaDetalheModelList: vendaDetalheModelListClone(vendaDetalheModelList!),
			vendaFreteModelList: vendaFreteModelListClone(vendaFreteModelList!),
		);			
	}

	vendaDetalheModelListClone(List<VendaDetalheModel> vendaDetalheModelList) { 
		List<VendaDetalheModel> resultList = [];
		for (var vendaDetalheModel in vendaDetalheModelList) {
			resultList.add(
				VendaDetalheModel(
					id: vendaDetalheModel.id,
					idVendaCabecalho: vendaDetalheModel.idVendaCabecalho,
					idProduto: vendaDetalheModel.idProduto,
					quantidade: vendaDetalheModel.quantidade,
					valorUnitario: vendaDetalheModel.valorUnitario,
					valorSubtotal: vendaDetalheModel.valorSubtotal,
					taxaDesconto: vendaDetalheModel.taxaDesconto,
					valorDesconto: vendaDetalheModel.valorDesconto,
					valorTotal: vendaDetalheModel.valorTotal,
				)
			);
		}
		return resultList;
	}

	vendaFreteModelListClone(List<VendaFreteModel> vendaFreteModelList) { 
		List<VendaFreteModel> resultList = [];
		for (var vendaFreteModel in vendaFreteModelList) {
			resultList.add(
				VendaFreteModel(
					id: vendaFreteModel.id,
					idVendaCabecalho: vendaFreteModel.idVendaCabecalho,
					idTransportadora: vendaFreteModel.idTransportadora,
					responsavel: vendaFreteModel.responsavel,
					conhecimento: vendaFreteModel.conhecimento,
					placa: vendaFreteModel.placa,
					ufPlaca: vendaFreteModel.ufPlaca,
					seloFiscal: vendaFreteModel.seloFiscal,
					quantidadeVolume: vendaFreteModel.quantidadeVolume,
					marcaVolume: vendaFreteModel.marcaVolume,
					especieVolume: vendaFreteModel.especieVolume,
					pesoBruto: vendaFreteModel.pesoBruto,
					pesoLiquido: vendaFreteModel.pesoLiquido,
				)
			);
		}
		return resultList;
	}

	
}